core/phase.js

```diff
@@ -206,6 +234,8 @@ PhaseBaseRuntime.prototype.put = function(data, tags) {
   } else {
     this.tags = this.baseTags.clone();
   }
+  // TODO: This misses tags when they are set after calling put().
+  trace.enabled && flowItemPut(this, this.tags.tags);
```

Comment:
I was just guarding things that are non-trivial. I'll move the checks inside flowItem*.